federation/pkg/kubefed/init/init.go

```diff
@@ -160,6 +168,21 @@ func initFederation(cmdOut io.Writer, config util.AdminConfig, cmd *cobra.Comman
 	etcdPVCapacity := cmdutil.GetFlagString(cmd, "etcd-pv-capacity")
 	dryRun := cmdutil.GetDryRunFlag(cmd)
 	storageBackend := cmdutil.GetFlagString(cmd, "storage-backend")
+	apiServiceType := cmdutil.GetFlagString(cmd, "api-service-type")
+	if apiServiceType != serviceTypeLoadBalancer && apiServiceType != serviceTypeNodePort {
+		return fmt.Errorf("api-service-type should be either %s or %s", serviceTypeLoadBalancer, serviceTypeNodePort)
+	}
+	apiServerAdvertiseAddress := cmdutil.GetFlagString(cmd, "api-server-advertise-address")
+	if apiServiceType == serviceTypeNodePort {
+		if apiServerAdvertiseAddress == "" {
+			return fmt.Errorf("api-server-advertise-address should be specified when 'api-service-type=nodeport'")
+		}
+
+		ip := net.ParseIP(apiServerAdvertiseAddress)
+		if ip == nil {
+			return fmt.Errorf("Invalid address provided for api-server-advertise-address")
+		}
```

Comment:
if service type is nodeport, I guess it will have to be one of the node IPs, isnt it.. ?